Null component update

pandmitr opened the pull request:
Change list:
- Loc:: added;
- component cache added;
- component class added;
- perfect codestyle;
- phpDoc added;
- result_modifier.php added;
- component_epilog.php added;
- php >= 5.4 required.

By pandmitr

pandmitr committed:
Null component updated.

Change list:

- Loc:: added;
- component cache added;
- component class added;
- perfect codestyle;
- phpDoc added;
- result_modifier.php added;
- component_epilog.php added.

By pandmitr

pandmitr committed:
Null component fixed.

Change list:

- fix encoding.

By pandmitr

Manriel commented on local/components/codecraft/null/templates/.default/component_epilog.php:
```diff
@@ -0,0 +1,23 @@
+ * @global CDatabase            $DB
+ */
+
+$APPLICATION->SetTitle($arResult['TITLE']);
```
If template code returns from cache, $arResult['TITLE'] == NULL, that is incorrect.

pandmitr replied:
Yes, it's possible to bx <= 10

Now it problem is not actual, see http://dev.1c-bitrix.ru/learning/course/?COURSE_ID=43&LESSON_ID=2975

pandmitr replied:
Sorry, you right. And I has any error, all fixed.

Manriel replied:
Yep. :-)
arResult array key must contains in SetResultCacheKeys method parameters to save in cache. This key do not included into cached keys, so, it do not saved in cache.
You can check it.

pandmitr replied:
Already fixed, see it

pandmitr committed:
Null component fixed.

Change list:

- fix component error;;
- fix result_modifier cache variables.

By pandmitr
